packages/dicomImageLoader/src/imageLoader/internal/rangeRequest.ts

```diff
@@ -0,0 +1,241 @@
+        done,
+        isLossy: !!retrieveOptions.isLossy,
+        loadNextRange: done
+          ? undefined
+          : async () => {
+              const loadedBytes = streamCache[imageId].byteArray.length;
+              const totalBytes = streamCache[imageId].totalBytes;
+              if (loadIsComplete(imageId)) {
+                return {
+                  complete: true,
+                  imageFrame: streamCache[imageId].byteArray,
+                };
+              }
+
+              const rangesFetched = streamCache[imageId].rangesFetched;
+              const rangeEnd =
+                Math.ceil(
+                  (totalBytes - loadedBytes) /
+                    ((totalRanges as number) - rangesFetched)
+                ) + loadedBytes;
+
+              const { bytes } = await fetchRangeAndAppend(
+                url,
+                imageId,
+                headers,
+                [loadedBytes, rangeEnd]
+              );
+
+              return {
+                complete: loadIsComplete(imageId),
+                imageFrame: bytes,
+                contentType,
+              };
+            },
+      });
+    } catch (err: any) {
+      errorInterceptor(err);
+      console.error(err);
+      reject(err);
+    }
+  });
+
+  return promise;
+}
+
+function loadIsComplete(imageId) {
+  return (
+    streamCache[imageId].byteArray.length === streamCache[imageId].totalBytes
+  );
+}
+
+async function fetchRangeAndAppend(
+  url: string,
+  imageId: string,
+  headers: any,
+  range: [number, number]
+) {
+  headers = Object.assign(headers, {
+    Range: `bytes=${range[0]}-${range[1]}`,
+  });
+  const response = await fetch(url, {
+    headers,
+    signal: undefined,
+  });
+
+  const responseArrayBuffer = await response.arrayBuffer();
+  const responseTypedArray = new Uint8Array(responseArrayBuffer);
+  const responseHeaders = response.headers;
+
+  // Append new data
+  const existingBytesForImageId = streamCache[imageId].byteArray;
+  let newByteArray: Uint8Array;
+  if (existingBytesForImageId) {
+    newByteArray = new Uint8Array(
+      existingBytesForImageId.length + responseTypedArray.length
+    );
+    newByteArray.set(existingBytesForImageId, 0);
+    newByteArray.set(responseTypedArray, existingBytesForImageId.length);
+  } else {
+    newByteArray = new Uint8Array(responseTypedArray.length);
+    newByteArray.set(responseTypedArray, 0);
+  }
+  streamCache[imageId].byteArray = newByteArray;
+
+  const contentRange = response.headers.get('Content-Range');
+  if (contentRange) {
+    streamCache[imageId].totalBytes = Number(
+      responseHeaders.get('Content-Range').split('/')[1]
+    );
+  } else {
+    streamCache[imageId].totalBytes = newByteArray.length;
+  }
+
+  loadTracking[imageId] = {
+    total: Number(streamCache[imageId].totalBytes),
+    loaded: newByteArray.length,
+  };
+
+  streamCache[imageId].rangesFetched += 1;
+
+  return {
+    bytes: newByteArray,
```

Comment:
can we say `byteArray: newByteArray` instead? `byets` is too unusuall

Reply:
Using encodedBytes as that is what is used in the streaming data